themes/10up-block-theme/includes/core.php

```diff
@@ -0,0 +1,190 @@
+ */
+function theme_setup() {
+	add_theme_support( 'editor-styles' );
+	add_editor_style( '/dist/css/frontend.css' );
+	remove_theme_support( 'core-block-patterns' );
+}
+
+/**
+ * Enqueue scripts for front-end.
+ *
+ * @return void
+ */
+function scripts() {
+
+	wp_enqueue_script(
+		'frontend',
+		TENUP_BLOCK_THEME_TEMPLATE_URL . '/dist/js/frontend.js',
+		get_asset_info( 'frontend', 'dependencies' ),
+		get_asset_info( 'frontend', 'version' ),
+		[
+			'strategy' => 'defer',
+		]
+	);
+}
+
+
+/**
+ * Enqueue styles for front-end.
+ *
+ * @return void
+ */
+function styles() {
+	wp_enqueue_style(
+		'tenup-theme-styles',
+		TENUP_BLOCK_THEME_TEMPLATE_URL . '/dist/css/frontend.css',
+		[],
+		get_asset_info( 'frontend', 'version' )
+	);
+}
+
+/**
+ * Enqueue styles for editor only.
+ *
+ * @return void
+ */
+function editor_style_overrides() {
+	wp_enqueue_style(
+		'tenup-theme-editor-style-overrides',
+		TENUP_BLOCK_THEME_TEMPLATE_URL . '/dist/css/editor-style-overrides.css',
+		[],
+		TENUP_BLOCK_THEME_VERSION
+	);
+
+	wp_enqueue_script(
+		'tenup-theme-block-extensions',
+		TENUP_BLOCK_THEME_TEMPLATE_URL . '/dist/js/block-extensions.js',
+		get_asset_info( 'block-extensions', 'dependencies' ),
+		get_asset_info( 'block-extensions', 'version' ),
+		true
+	);
+}
+
+/**
+ * register all icons located in the dist/svg folder
+ */
+function register_all_icons() {
+	if ( ! function_exists( '\UIKitCore\Helpers\register_icons' ) ) {
+		return;
+	}
+
+	$icon_paths = glob( TENUP_BLOCK_THEME_DIST_PATH . 'svg/*.svg' );
+	$icons      = array_map(
+		function ( $icon_path ) {
+			$icon_name = preg_replace( '#\..*$#', '', basename( $icon_path ) );
+
+			return new \UIKitCore\Icon(
+				$icon_name,
+				ucwords( str_replace( '-', ' ', $icon_name ) ),
+				$icon_path
+			);
+		},
+		$icon_paths
+	);
+
+	\UIKitCore\Helpers\register_icons(
+		[
+			'name'  => 'tenup',
+			'label' => 'Theme Icons',
+			'icons' => $icons,
+		]
+	);
+}
+
+/**
+ * Handles JavaScript detection.
+ *
+ * Adds a `js` class to the root `<html>` element when JavaScript is detected.
+ *
+ * @return void
+ */
+function js_detection() {
+
+	echo "<script>(function(html){html.className = html.className.replace(/\bno-js\b/,'js')})(document.documentElement);</script>\n";
+}
+
+/**
+ * Handles scrollbar width detection.
+ *
+ * Adds a JavaScript event listener to the DOMContentLoaded event. When the DOM is fully loaded,
+ * it calculates the width of the scrollbar and sets a CSS variable `--wp--custom--scrollbar-width` with the width.
+ * It also adds an event listener to the window resize event to update the scrollbar width when the window is resized.
+ *
+ * @return void
+ */
+function scrollbar_detection() {
+	echo '<script>window.addEventListener("DOMContentLoaded",()=>{const t=()=>window.innerWidth-document.body.clientWidth;const e=()=>{document.documentElement.style.setProperty("--wp--custom--scrollbar-width",`${t()}px`)};e();window.addEventListener("resize",e);});</script>' . "\n";
```

Comment:
Not sure we want this in the scaffold.

Reply:
[Here is more context](https://fueled10up.slack.com/archives/C10D34T33/p1680624401811279) 

CC @dmtrmrv